src/domain/ecoverse/ecoverse.service.ts

```diff
@@ -127,14 +139,26 @@ export class EcoverseService {
 
       // this.eventDispatcher.dispatch(events.ecoverse.query, { ecoverse: ecoverse });
       // Convert groups array into IGroups array
-      const groups: IUserGroup[] = [];
       if (!ecoverse.groups) {
         throw new Error('Unreachable');
       }
-      for (const group of ecoverse.groups) {
-        groups.push(group);
+      return ecoverse.groups as IUserGroup[];
+    } catch (e) {
+      // this.eventDispatcher.dispatch(events.logger.error, { message: 'Something went wrong in getMembers()!!!', exception: e });
+      throw e;
+    }
+  }
+
+  async getChallenges(): Promise<IChallenge[]> {
+    try {
+      const ecoverse: IEcoverse = await this.getEcoverse();
+
+      // this.eventDispatcher.dispatch(events.ecoverse.query, { ecoverse: ecoverse });
+      // Convert groups array into IGroups array
+      if (!ecoverse.challenges) {
+        throw new Error('Unreachable');
       }
-      return groups;
+      return ecoverse.challenges as IChallenge[];
```

Comment:
I'd suggest refactoring IEcoverse to have IChallenge[] instead of IChallenge[] | undefined member - and maybe in general everywhere where we have arrays. We don't seem to use undefined much.

Reply:
It cannot have this as we cannot set this field in the constructor due to restrictions from the ORM